java/flight/flight-core/src/main/java/org/apache/arrow/flight/auth2/ClientIncomingAuthHeaderMiddleware.java

```diff
@@ -23,39 +23,48 @@
 import org.apache.arrow.flight.CallInfo;
 import org.apache.arrow.flight.CallStatus;
 import org.apache.arrow.flight.FlightClientMiddleware;
+import org.apache.arrow.flight.grpc.CredentialCallOption;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * Middleware for capturing bearer tokens sent back from the Flight server.
  */
-public class ClientBearerTokenMiddleware implements FlightClientMiddleware {
-  private static final Logger logger = LoggerFactory.getLogger(ClientBearerTokenMiddleware.class);
+public class ClientIncomingAuthHeaderMiddleware implements FlightClientMiddleware {
+  private static final Logger logger = LoggerFactory.getLogger(ClientIncomingAuthHeaderMiddleware.class);
 
   private final Factory factory;
 
   /**
    * Factory used within FlightClient.
    */
   public static class Factory implements FlightClientMiddleware.Factory {
-    private final AtomicReference<String> bearerToken = new AtomicReference<>();
+    private final ClientHeaderHandler headerHandler;
+    private final AtomicReference<CredentialCallOption> credentialCallOption = new AtomicReference<>();
```

Comment:
I don't think this has to be an AtomicReference since this looks to be used single-threaded only.

Reply:
Done, updated.